arctic/chunkstore/date_chunker.py

```diff
@@ -74,7 +74,17 @@ def to_start_end(self, data):
         return self._get_date_range(data)
 
     def to_mongo(self, range_obj):
-        return range_obj.mongo_query()
+        if range_obj.start and range_obj.end:
+            return {'$and': [{'start': {'$lte': range_obj.end}}, {'end': {'$gte': range_obj.start}}]}
+        elif range_obj.start:
+            return {'end': {'$gte': range_obj.start}}
+        elif range_obj.end:
+            return {'start': {'$lte': range_obj.end}}
+        else:
+            return {}
 
     def filter(self, data, range_obj):
-        return data.ix[range_obj[0]:range_obj[1]]
+        return data.ix[range_obj.start:range_obj.end]
```

Comment:
seems to work fine with every range - no need to call that API. I added some unit tests for it.